recipes/gc/all/conanfile.py

```diff
@@ -0,0 +1,142 @@
+        ("disclaim",                    True,),
+        ("large_config",                True,),
+        ("gc_assertions",               False,),
+        ("mmap",                        False,),
+        ("munmap",                      True,),
+        ("dynamic_loading",             True,),
+        ("register_main_static_data",   True,),
+        ("gc_assertions",               False,),
+        ("checksums",                   False,),
+        ("single_obj_compilation",      False,),
+    )
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    for option, default in _autotools_options_defaults:
+        options[option] = [True, False]
+        default_options[option] = default
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if tools.Version(self.version) <= "8.0.4":
+            del self.options.throw_bad_alloc_library
+
+    def requirements(self):
+        if self.settings.os == "Windows":
+            self.requires("libatomic_ops/7.6.10")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("gc-{}".format(self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        for option, _ in self._autotools_options_defaults:
+                self._cmake.definitions["enable_{}".format(option)] = self.options.get_safe(option)
+        self._cmake.definitions["disable_gc_debug"] = not self.options.gc_debug
+        self._cmake.definitions["disable_handle_fork"] = not self.options.handle_fork
+        self._cmake.definitions["install_headers"] = True
+        self._cmake.definitions["build_tests"] = False
+        if tools.is_apple_os(self.settings.os):
+            cmake_osx_arch = {
+                "x86": "i386",
+            }.get(str(self.settings.arch), str(self.settings.arch))
+            self._cmake.definitions["CMAKE_OSX_ARCHITECTURES"] = cmake_osx_arch
+        self._cmake.verbose = True
+        self._cmake.parallel = False
+        self._cmake.configure()
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _extract_copyright(self):
+        readme_md = open(os.path.join(self._source_subfolder, "README.md")).read()
+        copyright_header = "## Copyright & Warranty\n"
+        index = readme_md.find(copyright_header)
+        assert index > 0
```

Comment:
```suggestion
        if index == 0:
            raise ConanException("Could not extract license from README file.")
```

You need to import ConanException too.

Reply:
`find` returns `-1` when a string cannot be found.